add callout for net new snapshots (#6521)
dbt-labs/docs.getdbt.com: website/docs/reference/snapshot-configs.md

<VersionBlock lastVersion="1.8">

> Starting from Versionless and Core v1.9, defining snapshots in a `.sql` file using a config block is a legacy method. You can define snapshots in YAML format using the latest [snapshot-specific configurations](/docs/build/snapshots#snapshot-meta-fields). For new snapshots, we recommend using these latest configs. If applying them to existing snapshots, you'll need to migrate over.

Reviewer: ```suggestion
Starting from Versionless and Core v1.9, defining snapshots in a `.sql` file using a config block is a legacy method. You can define snapshots in YAML format using the latest [snapshot-specific configurations](/docs/build/snapshot-configs#snapshot-specific-configurations). For new snapshots, we recommend using these latest configs. If applying them to existing snapshots, you'll need to migrate over.
```

Reviewer: Since this same text appears 3 times, would this be a good candidate for a snippet?

> Starting from Versionless and Core v1.9, defining snapshots in a `.sql` file using a config block is a legacy method. You can define snapshots in YAML format using the latest [snapshot-specific configurations](/docs/build/snapshot-configs#snapshot-specific-configurations). For new snapshots, we recommend using these latest configs. If applying them to existing snapshots, you'll need to migrate over.

Reviewer: Including this section for v1.8 creates awkwardness because then the "The following resource-specific configurations are only available to Snapshots:" sentence is just hanging there:

<img width="600" alt="image" src="https://github.com/user-attachments/assets/c5e4de00-52a5-45c3-a958-143815e7fe1c">

---

Do you have any ideas to make it look better for v1.8?

One idea: put it _within_ the "Config block" tab.

    [+](/reference/resource-configs/plus-prefix)[strategy](/reference/resource-configs/strategy): timestamp | check
    [+](/reference/resource-configs/plus-prefix)[updated_at](/reference/resource-configs/updated_at): <column_name>
    [+](/reference/resource-configs/plus-prefix)[check_cols](/reference/resource-configs/check_cols): [<column_name>] | all
    [+](/reference/resource-configs/plus-prefix)[invalidate_hard_deletes](/reference/resource-configs/invalidate_hard_deletes) : true | false    
    [+](/reference/resource-configs/plus-prefix)[snapshot_meta_column_names](/reference/resource-configs/snapshot_meta_column_names): {<dictionary>}

>     [+](/reference/resource-configs/plus-prefix)[dbt_valid_to_current](/reference/resource-configs/dbt_valid_to_current): <string> 

Reviewer 2: `dbt_valid_to_current` config is available in 1.9+

Reviewer 2: Oh maybe I'm not reading these version blocks right?

Reviewer: @graciegoheen You gotta expand the view so you can see line 67 / 64:

<img width="939" alt="image" src="https://github.com/user-attachments/assets/c2118e78-e1c4-4883-a169-24e9f3e9f1e3">

<VersionBlock firstVersion="1.9">

> Starting from Versionless and Core v1.9, defining snapshots in a `.sql` file using a config block is a legacy method. You can define snapshots in YAML format using the latest [snapshot-specific configurations](/docs/build/snapshots#snapshot-meta-fields). For new snapshots, we recommend using these latest configs. If applying them to existing snapshots, you'll need to migrate over.

Reviewer: ```suggestion
Starting from Versionless and Core v1.9, defining snapshots in a `.sql` file using a config block is a legacy method. You can define snapshots in YAML format using the latest [snapshot-specific configurations](/docs/build/snapshot-configs#snapshot-specific-configurations). For new snapshots, we recommend using these latest configs. If applying them to existing snapshots, you'll need to migrate over.
```

<VersionBlock firstVersion="1.9">

> Configurations can be applied to snapshots using the [YAML syntax](/docs/build/snapshots#configuring-snapshots), available in Versionless and dbt v1.9 and higher, in the `snapshot` directory file.

Reviewer 2: It's not _just_ configuration - it's the entire snapshot that can be defined in YAML now

</Tabs>

### Snapshot configuration migration

The latest snapshot configurations introduced in dbt v1.9 (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names), [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current), and `hard_deletes`) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies in your snapshots:

#### For existing snapshots
- Migrate tables &mdash; Migrate the previous snapshot to the new table schema and values:
  - Create a backup copy of your snapshots.
  - Use `alter` statements as needed (or a script to apply `alter` statements) to ensure table consistency.
- New configurations &mdash; Convert the configs one at a time, testing as you go. 

:::warning
If you use one of the latest configs, such as `dbt_valid_to_current`, without migrating your data, you may have mixed old and new data, leading to an incorrect downstream result.

> :::

Reviewer: I think we should link to this section within the v1.9 upgrade guide:

website/docs/docs/dbt-versions/core-upgrade/06-upgrading-to-v1.9.md

</Tabs>

### Snapshot configuration migration

> The latest snapshot YAML configurations introduced in dbt v1.9 (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) and [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current)) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies in your snapshots:

Reviewer 2: I think it's a short enough list we can just name them all - `snapshot_meta_column_names`,  `dbt_valid_to_current`, `hard_deletes`

Reviewer 2: Also, you say "For existing snapshots, we recommend the following" but then go into a list for new snapshots?

Author: good call! I'll swap so it flows better 🙏

> The latest snapshot YAML configurations introduced in dbt v1.9 (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) and [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current)) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies in your snapshots:

Reviewer 2: ```suggestion
The latest snapshot configurations introduced in dbt v1.9 (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) and [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current)) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies in your snapshots:
```

> The latest snapshot YAML configurations introduced in dbt v1.9 and higher (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) and [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current)) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies to your snapshots:

Author: note i'm using 'latest' so it's more scalable for the long term. if i use 'new', it'll become outdated eventually i think.

> The latest snapshot YAML configurations introduced in dbt v1.9 and higher (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) and [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current)) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies to your snapshots:

Reviewer 3: ```suggestion
The latest snapshot YAML configurations introduced in dbt v1.9 (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) and [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current)) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies in your snapshots:
```

#### For new snapshots

> - Use the latest snapshot YAML configurations when creating new snapshots that didn't exist previously.

Reviewer 2: I think we can just delete this?

#### For existing snapshots
- Migrate tables &mdash; Migrate the previous snapshot to the new table schema and values:
  - Create a backup copy of your snapshots.

>   - Use `alter` statements as needed to ensure table consistency or use a script to apply the `alter` statements.

Reviewer 3: ```suggestion
  - Use `alter` statements as needed (or a script to apply `alter` statements) to ensure table consistency.
```

> - New YAML config &mdash; Convert the YAML configurations one at a time, testing as you go. 

Reviewer 2: We can just call these "new configs" we don't need to say YAML configs

> If you use one of the latest configs, such as `dbt_valid_to_current`, without migrating your data, you may have mixed old and new data, leading to an incorrect downstream result.

Reviewer 2: should this be in a warning callout?

> If you use one of the latest configs, such as `dbt_valid_to_current`, without migrating your data may result in mixed old and new data, leading to incorrect downstream result.

Reviewer 3: ```suggestion
If you use one of the latest configs, such as `dbt_valid_to_current`, without migrating your data, you may have mixed old and new data, leading to an incorrect downstream result.
```

<ConfigGeneral />

> :::tip

Reviewer 3: Should we make this a snippet since its across three different sections?

> Configuring snapshots in YAML is recommended for new snapshot configurations. If you have existing snapshots using the `.sql` config, they'll continue to work as expected. When you're ready to migrate to YAML, we recommend:

Reviewer 2: ```suggestion
Configuring snapshots in YAML is recommended for new snapshots. If you have existing snapshots using the `.sql` config, they'll continue to work as expected. When you're ready to migrate to YAML, we recommend:
```

> 1. Creating a backup copy of your snapshots (or using version control)

Reviewer 2: By using dbt, aren't you already using version control?

2. Converting the configurations one at a time, testing as you go

> 3. Using `alter` statements as needed to ensure table consistency or using a script to apply the `alter` statements

Reviewer 2: This shouldn't be needed if you're just moving from sql -> yml, right? Is this just if you _add_ one (or more) of the new configs?

Author: updated this to the below:

### Snapshot configuration migration

The latest snapshot YAML configurations introduced in dbt v1.9 and higher (such as [`snapshot_meta_column_names`](/reference/resource-configs/snapshot_meta_column_names) and [`dbt_valid_to_current`](/reference/resource-configs/dbt_valid_to_current)) are best suited for new snapshots. For existing snapshots, we recommend the following to avoid any inconsistencies to your snapshots:

#### For new snapshots
- Use the latest snapshot YAML configurations when creating new snapshots that didn't exist previously.

#### For existing snapshots
- Migrate tables &mdash; Migrate the previous snapshot to the new table schema and values:
  - Create a backup copy of your snapshots.
  - Use `alter` statements as needed to ensure table consistency or use a script to apply the `alter` statements.
- New YAML config &mdash; Convert the YAML configurations one at a time, testing as you go. 

If you use one of the latest configs, such as `dbt_valid_to_current`, without migrating your data may result in mixed old and new data, leading to incorrect downstream result.

<VersionBlock firstVersion="1.9">

> Starting from Versionless and Core v1.9, defining snapshots in a `.sql` file using a config block is a legacy method. You can define snapshots in YAML format using the latest [configurations](/docs/build/snapshots#configuring-snapshots). You can also continue using the legacy method for existing snapshots, but we recommend migrating over to the latest YAML format for better readability and maintainability.

Reviewer: Do we want to align this section with the other ones above? Or do we want it to be slightly different?

If we want it to be the same, then we'd update it like this:
```suggestion
Starting from Versionless and Core v1.9, defining snapshots in a `.sql` file using a config block is a legacy method. You can define snapshots in YAML format using the latest [snapshot-specific configurations](/docs/build/snapshot-configs#snapshot-specific-configurations). For new snapshots, we recommend using these latest configs. If applying them to existing snapshots, you'll need to migrate over.
```
